grpc/server.go

```diff
@@ -43,8 +44,27 @@ func (s *Server) StreamSearch(req *v1.SearchRequest, ss v1.WebserverService_Stre
 		return status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	onMatch := stream.SenderFunc(func(res *zoekt.SearchResult) {
-		ss.Send(res.ToProto())
+	onMatch := stream.SenderFunc(func(r *zoekt.SearchResult) {
+		result := r.ToProto()
+
+		f := func(filesChunk []*v1.FileMatch) error {
+			return ss.Send(&v1.SearchResponse{
+				Stats:         result.GetStats(),
+				Progress:      result.GetProgress(),
+				RepoUrls:      result.GetRepoUrls(),
+				LineFragments: result.GetLineFragments(),
+
+				Files: filesChunk,
+			})
+		}
+
+		chunker := chunk.New(f)
```

Comment:
shouldn't chunker live outside of f so it can re-use the buffer/etc. You just much call Flush on each event?

Reply:
I'm not quite sure what you mean, but chunker does live outside of `f`. Are you talking about it living outside of `onMatch` entirely?

Each of the search response structs have progress / stats that are unique for that particular match. If the chunker lived outside of onmatch, then we'd have to deal with combining `stats` and `progress` across multiple matches - which didn't seem correct to me.